custom/icds_reports/sqldata/exports/dashboard_usage.py

```diff
@@ -0,0 +1,209 @@
+        return location_list.append('awc_id')
+
+    def convert_boolean_to_string(self, value):
+        return 'yes' if value else 'no'
+
+    def get_users_by_location(self, user_supported_locations):
+        user_query = UserES().mobile_users().domain(self.domain).location(
+            user_supported_locations
+        ).fields(['username', 'assigned_location_ids', 'last_login'])
+        return [u for u in user_query.run().hits]
+
+    def get_names_from_ids(self, location_id):
+        if location_id == 'doc_id':
+            return 'awc_name'
+        else:
+            return location_id.split('_')[0] + '_name'
+
+    def convert_rs_to_matrix(self, results_queryset, main_location_type):
+        """
+
+        :param results_queryset: AwcLocation values queryset
+        :param main_location_type: parent location type to fetch only locations below that level
+        :return: location matrix similar to a dataframe and locations ids of all the locations below the given
+         location
+        """
+        location_matrix = []
+        location_ids = []
+        sub_location_types = self.location_types[self.location_types.index(main_location_type) + 1:]
+        for result in results_queryset:
+            row_data = [result['state_id'], result['district_id'], result['block_id'], result['supervisor_id'],
+                        result['doc_id']]
+            location_matrix.append(row_data)
+            # adding only descendants to the main location
+            for sub_location in sub_location_types:
+                if result[sub_location] not in location_ids:
+                    location_ids.append(result[sub_location])
+        return location_matrix, location_ids
+
+    def get_location_id_string_from_location_type(self, location_type):
+        if location_type == 'awc':
+            location_id = 'doc_id'
+        else:
+            location_id = location_type + '_id'
+        return location_id
+
+    def get_location_type_string_from_location_id(self, location_id):
+        return location_id.replace('_id', '')
+
+    def check_if_date_in_last_week(self, date):
+        if date is None:
+            return False
+        d = datetime.datetime.strptime(date, "%Y-%m-%d")
+        now = datetime.datetime.now()
+        return (d - now).days < 7
+
+    def prepare_is_launched_agg_list(self, location_type, location_id):
+        '''
+        populates the location_id vs is_launched status and location_id vs location_type mappings for the
+         descendants of given location
+        :param location_type:
+        :param location_id:
+        :return: None
+        '''
+        location_type = self.get_location_id_string_from_location_type(location_type)
+        sub_location_types = self.agg_required_fields[self.location_types.index(location_type) + 1:]
+        filter_dict = {location_type: location_id}
+        aggregate_records = AggAwc.objects.filter(**filter_dict).values(*sub_location_types)
```

Comment:
This is literally, fetching all rows for all the past months till nowfor that location and on every aggregate level which is not required and this could be very slow.

Reply:
added